runtime/testy/stateManager.py

```diff
@@ -32,12 +32,52 @@ def addPipe(self, processName, pipe):
     self.processMapping[processName] = pipe
     pipe.send(RUNTIME_CONFIG.PIPE_READY.value)
 
-  def getValue(self, key):
-    self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(self.state[key])
+  def getValue(self, keys):
+    result = self.state
+    i = 0
```

Comment:
The pythonic way of doing iterations like this is to use a for loop rather than a while loop. 
Cleaner, more readable, less likely to have an error when you forget `i+=1`

In fact, in general it's better to use a for loop over a while loop when possible. 

for key in keys: 
  result = result[key]